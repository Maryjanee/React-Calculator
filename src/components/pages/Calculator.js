/* eslint-disable import/extensions */
import React, { useState } from 'react';
import ButtonPanel from '../ButtonPanel';
import Display from '../Display';
import '../../styles/App.css';
import calculate from '../../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: null, next: null, operation: null });

  const handleClick = buttonName => {
    const { total, next, operation } = state;
    setState(calculate({ total, next, operation }, buttonName));
  };

  const { total, next } = state;
  return (
    <div className="calculator">
      <Display result={next || total || 0} />
      <ButtonPanel onClick={handleClick} />
    </div>
  );
};

export default Calculator;
