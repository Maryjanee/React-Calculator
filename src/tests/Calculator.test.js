import TestRenderer from 'react-test-renderer';
import Calculator from '../components/pages/Calculator';

it('Snapshot test for Calculator component', () => {
    const CalculatorShot = TestRenderer
    .create(<Calculator />)
    .toJSON();
  expect(CalculatorShot).toMatchSnapshot();
}) 