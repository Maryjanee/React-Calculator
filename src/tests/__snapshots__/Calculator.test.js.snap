// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Calculator component 1`] = `
<div
  className="calculator"
>
  <div
    className="display-container"
  >
    <p
      className="display-value"
    >
      0
    </p>
  </div>
  <div
    className="btn-panel"
  >
    <div
      className="btn-group"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="btn-group"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        *
      </button>
    </div>
    <div
      className="btn-group"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="btn-group"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="btn-group last"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
